iroh-relay/src/dns.rs

```diff
@@ -250,12 +251,55 @@ impl DnsResolver {
     }
 }
 
+impl Default for DnsResolver {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
 impl From<TokioResolver> for DnsResolver {
     fn from(resolver: TokioResolver) -> Self {
         DnsResolver(resolver)
     }
 }
 
+/// TXT records returned from [`DnsResolver::lookup_txt`]
+#[derive(Debug, Clone)]
+pub struct TxtLookup(pub(self) hickory_resolver::lookup::TxtLookup);
```

Comment:
TIL about `pub(self)` (had to look this up in https://doc.rust-lang.org/reference/visibility-and-privacy.html#r-vis.scoped.self).  I'm confused why this even exists in rust?  Seems like leaving it off is exactly the same, so why do folks want to spell it out sometimes?  What is the motivation here?

Reply:
Yea, dunno, removed it :)